src/main/java/school/hei/haapi/service/PaymentService.java

```diff
@@ -88,8 +89,28 @@ List<Payment> getByStudentIdAndFeeId(String studentId, String feeId) {
 
   public void computeRemainingAmount(String feeId, int amount) {
     Fee associatedFee = feeService.getById(feeId);
+    User student = associatedFee.getStudent();
+    User.Status[] statuses = new User.Status[2];
+    statuses[0] = student.getStatus();
+
     associatedFee.setRemainingAmount(associatedFee.getRemainingAmount() - amount);
-    computeUserStatusAfterPayingFee(associatedFee.getStudent());
+    computeUserStatusAfterPayingFee(student);
+    statuses[1] = student.getStatus();
+    if (SUSPENDED.equals(statuses[0]) && ENABLED.equals(statuses[1])) {
+      Payment payment =
+          Payment.builder()
+              // TIPS: no need  .type()
+              .fee(associatedFee)
+              .amount(amount)
+              .creationDatetime(Instant.now())
+              .comment(associatedFee.getComment())
+              .build();
+      SuspensionEndedEmailBody suspensionEndedEmailBody = SuspensionEndedEmailBody.from(payment);
+      eventProducer.accept(List.of(suspensionEndedEmailBody));
+      log.info(
+          "End of suspension notification for user {} sent to Queue.",
+          suspensionEndedEmailBody.getMpbsAuthorEmail());
```

Comment:
make this logic for sending mail in a separate method as notifyStudentForEnabling or something like that